src/utils/hooks/api/loans/get-usd-values.ts

```diff
@@ -0,0 +1,38 @@
+import { CurrencyIdLiteral, LendPosition, LoanPosition } from "@interlay/interbtc-api";
+import Big from "big.js";
+
+import { Prices } from "@/common/types/util.types";
+import { convertMonetaryAmountToValueInUSD } from "@/common/utils/utils";
+import { getTokenPrice } from "@/utils/helpers/prices";
+
+const getTotalEarnedInterestUSDValue = (lendPositions: LendPosition[], prices: Prices): Big => {
+    return lendPositions.reduce((totalValue: Big, position: LendPosition) => {
+        const { currency, earnedInterest } = position;
+        // TODO: Remove type casting after useGetPrices hook is refactored
+        const price = getTokenPrice(prices, currency.ticker as CurrencyIdLiteral)?.usd;
+
+        if (price === undefined) {
+            console.error(`useGetAccountCollateralization: No exchange rate found for currency: ${currency.name}`);
+        }
+
```

Comment:
I think this is a better approach.

```
if(!positionUSDValue) {
  return totalValue;
}

const positionUSDValue = convertMonetaryAmountToValueInUSD(earnedInterest, price);
return totalValue.add(positionUSDValue);
```

Reply:
Good suggestion, I simplified it with safer check and ternary operator:
```
return positionUSDValue === null ? totalValue : totalValue.add(positionUSDValue);
```